File: LAB2/tailfact.c
#include<stdio.h>

int tailfact(int a, long int n)
{
    if(n == 0)
       return a;
    else 
       return tailfact(n * a, n - 1);   
}

int main()
{
    long int num, result;
    printf("Enter a number: ");
    scanf("%d", &num);
    result = tailfact(1, num);
    printf("The factorial of %d is %d.\n", num, result);
    return 0;
}

File: 2.Recursion/TailFact/tailfact.py
def tailfact(a, n):
    if n == 0:
        return a
    else:
        return tailfact(n * a, n - 1)

num = int(input("Enter a number: "))
result = tailfact(1, num)
print(f"The factorial of {num} is {result}.")